src/main/java/org/ngafid/common/terrain/TerrainCache.java

```diff
@@ -13,14 +12,14 @@ public final class TerrainCache {
     private static final Logger LOG = Logger.getLogger(TerrainCache.class.getName());
     
 
-    // Load factor is arbitrary because we always remove the eldest entry
-    private static final Map<Coordinate, SRTMTile> TILE_CACHE = new ConcurrentHashMap<>(new LinkedHashMap<>(MAX_CACHE_SIZE, 0.75f, true) {
-        @Override
-        protected boolean removeEldestEntry(Map.Entry<Coordinate, SRTMTile> eldest) {
-            return size() > MAX_CACHE_SIZE;
-        }
-    });
-    
+     private static final ThreadLocal<LinkedHashMap<Coordinate, SRTMTile>> THREAD_LOCAL_TILE_CACHE =
+        // Load factor is arbitrary because we always remove the eldest entry
+        ThreadLocal.withInitial(() -> new LinkedHashMap<>(MAX_CACHE_SIZE, 0.75f, true) {
+            @Override
+            protected boolean removeEldestEntry(Map.Entry<Coordinate, SRTMTile> eldest) {
+                return size() > MAX_CACHE_SIZE;
```

Comment:
```
if (MAX_CACHE_SIZE > 0)
    return size() > MAX_CACHE_SIZE;
else
    return false;
````

To allow for an unbounded cache (useful in prod).

Reply:
You got it chief